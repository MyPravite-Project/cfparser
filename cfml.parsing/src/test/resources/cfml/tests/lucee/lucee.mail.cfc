component{
  function foo(){
    mail from="test@test.de", to="test@test.de";
  }
}