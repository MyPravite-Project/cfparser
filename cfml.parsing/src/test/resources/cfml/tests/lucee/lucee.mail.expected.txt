/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
MAIL                     <mail>
IDENTIFIER               <from>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <test@test.de>
CLOSE_STRING             <">
','                      <,>
TO                       <to>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <test@test.de>
CLOSE_STRING             <">
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (tagOperatorStatement
                (cfmlfunctionStatement
                  (cfmlFunction mail)
                  (paramStatementAttributes
                    (param
                      (multipartIdentifier (identifier from))
                      =
                      (startExpression
                        (baseExpression
                          (unaryExpression
                            (primaryExpression
                              (literalExpression (stringLiteral " (stringLiteralPart test@test.de) "))
                            )
                          )
                        )
                      )
                    )
                    ,
                    (param
                      (multipartIdentifier (identifier to))
                      =
                      (startExpression
                        (baseExpression
                          (unaryExpression
                            (primaryExpression
                              (literalExpression (stringLiteral " (stringLiteralPart test@test.de) "))
                            )
                          )
                        )
                      )
                    )
                  )
                  ;
                )
              )
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function foo()     {
mail from='test@test.de' to='test@test.de';;

    }
}
/*======*/